py/airflow.py

```diff
@@ -0,0 +1,298 @@
+"""Utilities for working with Airflow."""
+import datetime
+import json
+import logging
+import os
+import sys
+import tempfile
+import time
+
+import google.auth
+import google.auth.transport.requests
+from google.cloud import storage  # pylint: disable=no-name-in-module
+from py import prow  # pylint: disable=ungrouped-imports
+from py import util  # pylint: disable=ungrouped-imports
+import requests
+
+E2E_DAG = "tf_k8s_tests"
+
+# Currently we don't use airflow/api/client because there's not actually much
+# functionality in the client. So rather than take a dependency on airflow
+# we just copied the code we need. We can revisit that if the Airflow client
+# adds more functionality
+#
+# Note Airflow's client uses urljoin which doesn't produce the expected
+# behavior when base_url; see
+# https://stackoverflow.com/questions/10893374/python-confusions-with-urljoin
+class AirflowClient(object):
+  """Extension of the Airflow client."""
+  def __init__(self, base_url, credentials=None, verify=True):
+    """Construct the client.
+
+    Args:
+      base_url: Typically something like https://host:port.
+      credentials: A google.auth.credentials object used to obtain bearer
+        tokens or None if no credentials are needed
+      verify: Whether or not to verify TLS certs
+    """
+    self._api_base_url = base_url
+    if self._api_base_url[-1] == "/":
+      self._api_base_url = self._api_base_url[:-1]
+    self._credentials = credentials
+    self._verify = verify
+
+  def _request(self, url, method='GET', json_body=None):
+    params = {
+          'url': url,
+          "headers": {},
+        }
+    if json is not None:
+      params['json'] = json_body
+    params["verify"] = self._verify
+    if self._credentials:
+      if not self._credentials.valid:
+        request = google.auth.transport.requests.Request()
+        self._credentials.refresh(request)
+      self._credentials.apply(params["headers"])
+    resp = getattr(requests, method.lower())(**params)  # pylint: disable=not-callable
+    if not resp.ok:
+      try:
+        data = resp.json()
+      # TODO(jlewi): This code was copied from Airflow client.
+      # We should see if we can catch a more specific exception.
+      except Exception:  # pylint: disable=broad-except
+        data = {}
```

Comment:
log here? just in case it's a 4xx error?

Reply:
Done